addons/purchase/views/purchase_views.xml

```diff
@@ -871,6 +873,18 @@
         </field>
     </record>
 
+    <record id="action_purchase_cancel" model="ir.actions.server">
+        <field name="name">Cancel</field>
+        <field name="sequence">1</field>
+        <field name="model_id" ref="purchase.model_purchase_order"/>
+        <field name="binding_model_id" ref="purchase.model_purchase_order"/>
+        <field name="state">code</field>
+        <field name="code">
+            if records:
+                action = records.button_cancel()
+        </field>
+    </record>
+
```

Comment:
As we have a button in the list view. Let's keep it simple for now and remove the server action

Reply:
Done removed server action.